design.md

```diff
@@ -2,12 +2,27 @@
 
 ## Who is the target for this design, e.g., are you assuming any knowledge on the part of the language users?
 
+This design is for CS5 students. We assume that students have a basic understanding of scope and syntax as well as reading the picobot assignment. At the very least, someone should be able to read a program and understand what steps picobot should take.
+
 ## Why did you choose this design, i.e., why did you think it would be a good idea for users to express the maze-searching computation using this syntax?
 
+We chose this design because they seemed like natural improvements over the original picobot syntax. We think it would be better for maze-searching computation using this syntax because it is easier to read and debug. Also, many of the repeated statements can be removed (i.e. the rules are grouped by state  instead of restated on each rule).
+
 ## What behaviors are easier to express in your design than in Picobot’s original design?  If there are no such behaviors, why not?
 
+It is easier in our design to write complex rules because of the better readability.
+Also, we've designed a way to give orderings to rules (if-elseif-else style) which may allow users to write more interesting behaviors with fewer states. 
```

Comment:
Perhaps this is what you were getting at with 'complex rules', but I'd say your design helps express rules with high degrees of overlap.